packages/ckit/src/providers/mercury/MercuryProvider.ts

```diff
@@ -24,21 +24,44 @@ function toCell(resolved: ResolvedOutpoint): Cell {
   };
 }
 
+interface BatchRequest {
+  method: string;
+  params: unknown[];
+}
+
 export class MercuryProvider extends AbstractProvider {
   readonly mercury: MercuryClient;
   readonly rpc: RPC;
+  readonly rpcUrl: string;
 
-  constructor(
-    mercuryRpc: MercuryClient | string = 'http://127.0.0.1:8116',
-    ckbRpc: RPC | string = 'http://127.0.0.1:8114',
-  ) {
+  constructor(mercuryRpc = 'http://127.0.0.1:8116', ckbRpc = 'http://127.0.0.1:8114') {
     super();
 
-    if (mercuryRpc instanceof MercuryClient) this.mercury = mercuryRpc;
-    else this.mercury = new MercuryClient(mercuryRpc);
+    this.mercury = new MercuryClient(mercuryRpc);
+    this.rpc = new RPC(ckbRpc);
+    this.rpcUrl = ckbRpc;
+  }
+
+  /* eslint-disable @typescript-eslint/no-explicit-any */
+  async batchRequestCkb(request: BatchRequest): Promise<any> {
+    const batch = [];
+    for (let i = 0; i < request.params.length; i++) {
+      batch.push({
+        id: 42,
+        jsonrpc: '2.0',
+        method: request.method,
+        params: [request.params[i]],
+      });
+    }
+    const res = await fetch(this.rpcUrl, {
```

Comment:
the `fetch` is a browser API, it does not work in NodeJS

Reply:
since we depended on `@open-rpc/client-js`so indirectly depends on the `isomorphic-fetch`, but such a dependency is not safe, we still need to explicitly depend to ensure that the program can run as expected 

```ts
import fetch from 'isomorphic-fetch '
```
`package.json`
```json
... "isomorphic-fetch ": "^3.0.0"
```